File: app/src/main/java/net/radityalabs/contactapp/data/network/response/ContactDetailResponse.java
package net.radityalabs.contactapp.data.network.response;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by radityagumay on 4/15/17.
 */

public class ContactDetailResponse implements Parcelable {

    @SerializedName("id")
    @Expose
    public int id;
    @SerializedName("first_name")
    @Expose
    public String firstName;
    @SerializedName("last_name")
    @Expose
    public String lastName;
    @SerializedName("email")
    @Expose
    public String email;
    @SerializedName("phone_number")
    @Expose
    public String phoneNumber;
    @SerializedName("profile_pic")
    @Expose
    public String profilePic;
    @SerializedName("favorite")
    @Expose
    public boolean isFavorite;

    public ContactDetailResponse() {

    }

    protected ContactDetailResponse(Parcel in) {
        id = in.readInt();
        firstName = in.readString();
        lastName = in.readString();
        email = in.readString();
        phoneNumber = in.readString();
        profilePic = in.readString();
        isFavorite = in.readByte() != 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(id);
        dest.writeString(firstName);
        dest.writeString(lastName);
        dest.writeString(email);
        dest.writeString(phoneNumber);
        dest.writeString(profilePic);
        dest.writeByte((byte) (isFavorite ? 1 : 0));
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<ContactDetailResponse> CREATOR = new Creator<ContactDetailResponse>() {
        @Override
        public ContactDetailResponse createFromParcel(Parcel in) {
            return new ContactDetailResponse(in);
        }

        @Override
        public ContactDetailResponse[] newArray(int size) {
            return new ContactDetailResponse[size];
        }
    };
}

File: app/src/main/java/net/radityalabs/contactapp/data/network/response/ContactDetailResponse.kt
package net.radityalabs.contactapp.data.network.response

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created by radityagumay on 4/15/17.
 */

class ContactDetailResponse : Parcelable {

    @SerializedName("id")
    @Expose
    var id: Int = 0
    @SerializedName("first_name")
    @Expose
    var firstName: String
    @SerializedName("last_name")
    @Expose
    var lastName: String
    @SerializedName("email")
    @Expose
    var email: String
    @SerializedName("phone_number")
    @Expose
    var phoneNumber: String
    @SerializedName("profile_pic")
    @Expose
    var profilePic: String
    @SerializedName("favorite")
    @Expose
    var isFavorite: Boolean = false

    constructor() {

    }

    protected constructor(`in`: Parcel) {
        id = `in`.readInt()
        firstName = `in`.readString()
        lastName = `in`.readString()
        email = `in`.readString()
        phoneNumber = `in`.readString()
        profilePic = `in`.readString()
        isFavorite = `in`.readByte().toInt() != 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(id)
        dest.writeString(firstName)
        dest.writeString(lastName)
        dest.writeString(email)
        dest.writeString(phoneNumber)
        dest.writeString(profilePic)
        dest.writeByte((if (isFavorite) 1 else 0).toByte())
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {

        val CREATOR: Parcelable.Creator<ContactDetailResponse> = object : Parcelable.Creator<ContactDetailResponse> {
            override fun createFromParcel(`in`: Parcel): ContactDetailResponse {
                return ContactDetailResponse(`in`)
            }

            override fun newArray(size: Int): Array<ContactDetailResponse> {
                return arrayOfNulls(size)
            }
        }
    }
}
